identity-android/src/main/java/com/android/identity/android/legacy/Utility.java

```diff
@@ -403,8 +234,10 @@ byte[] encodeStaticAuthData(
 
             // Store issuerSignedMapping and issuerAuth (the MSO) in staticAuthData...
             //
-            byte[] staticAuthData = encodeStaticAuthData(
-                    issuerSignedMapping, encodedIssuerAuth);
+            byte[] staticAuthData = new StaticAuthDataGenerator()
+                    .setDigestIdMapping(issuerSignedMapping)
+                    .setEncodedIssuerAuth(encodedIssuerAuth)
+                    .generate();
```

Comment:
So, basically, the existing code in `Utility.java` uses a slightly different API than what we want to use for `StaticAuthDataParser` and `StaticAuthDataGenerator`. The existing code uses `IssuerSignedItem` as the `byte[]` values while our new API should use `IssuerSignedItemBytes`... as you can see in other comments, I pointed out the difference.

So I think we should just leave the existing API functions alone and not try to use it here. At some point we'll delete the stuff in `Utility.java`.

Reply:
Assuming I understood the intended format of `IssuerSignedItemMetadataBytes`, I found I was able to still fully replace the code in `Utility.java` since the generating function did create a tagged bytestring (even though it didn't expect it as an input) so the parsing function expected that format, which made the switch to `StaticAuthDataParser` quite easy still.